pin/pin_test.go

```diff
@@ -218,43 +199,3 @@ func TestFlush(t *testing.T) {
 		t.Fatal("expected key to still be pinned")
 	}
 }
-
-func TestPinRecursiveFail(t *testing.T) {
```

Comment:
this test should also stay. it's testing whether pinning a node that isn't there can be pinned. very much worth testing.